File: src/my_py_pkg/my_py_pkg/templates/subscriber.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node

from example_interfaces.msg import String # Depends on the format of messages from subscribed topics


class SubscriberNode(Node): #TODO

    def __init__(self):
        super().__init__("[NAME OF NODE]") #TODO
        
        self.subscriber_ = self.create_subscription(String, "[NAME OF TOPIC]", self.callback_topic, 10)  #TODO # Topic to subscribe to
        
        self.get_logger().info("[NAME OF NODE] has been started.")

    def callback_topic(self, msg):
        self.get_logger().info(msg.data) # prints the received messages


def main(args=None):
    rclpy.init(args=args)
    node = SubscriberNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()

File: src/my_cpp_pkg/src/templates/subscriber.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

class SubscriberNode: public rclcpp::Node //TODO (Ex: SmartphoneNode)
{
public:
    SubscriberNode(): Node("[NODE NAME]") //TODO (Ex: smartphone)
    {
        subscriber_ = this->create_subscription<example_interfaces::msg::String>(
            "[TOPIC NAME]", 10,
            std::bind(&SubscriberNode::callbackFunction, this, std::placeholders::_1)); // TODO: topic name and callback function (Ex: robot_news, callbackRobotNews)

        RCLCPP_INFO(this->get_logger(), "Subcriber has been started."); // Print statement
    }

private:
    void callbackFunction(const example_interfaces::msg::String::SharedPtr msg) //TODO (Ex: callbackRobotNews)
    {
        RCLCPP_INFO(this->get_logger(), "%s", msg->data.c_str());
    }

    rclcpp::Subscription<example_interfaces::msg::String>::SharedPtr subscriber_;

};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<SubscriberNode>(); //TODO (Ex: SmartphoneNode)
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}